tests/unit_tests/conftest.py

```diff
@@ -1677,3 +1710,46 @@ def helper(
         )
 
     yield helper
+
+
+class InjectedGraphQLRequestCreator(Protocol):
+    def __call__(
+        self,
+        body: Union[str, Exception] = "{}",
+        status: int = 200,
+        counter: int = -1,
+    ) -> InjectedResponse:
+        ...
+
+
+# Injected responses
+@pytest.fixture(scope="function")
+def inject_graphql_response(base_url: str) -> InjectedGraphQLRequestCreator:
+    def helper(
+        operation_name: str,
+        body: Union[str, Exception] = "{}",
+        status: int = 200,
+        counter: int = -1,
+    ) -> InjectedResponse:
+
+        if status > 299:
+            message = body if isinstance(body, str) else "::".join(body.args)
+            body = DeliberateHTTPError(status_code=status, message=message)
+
+        def predicate(request: requests.PreparedRequest) -> bool:
+            query = wandb_gql.gql(json.loads(request.body)["query"])
+            return query.definitions[0].name.value == operation_name
```

Comment:
does it make sense to use `partial` when passing this method instead of relaying on the scope? 
i.e. pass the `operation_name`

Reply:
IIUC, you're pointing at the problem where, when you dynamically define a function that refers to some variable in its enclosing scope, that variable can get a new value assigned to it before the function gets called? [This problem](https://stackoverflow.com/questions/3431676/creating-functions-or-lambdas-in-a-loop-or-comprehension)? I feel safe from that here because I can't imagine any circumstance under which `operation_name` would ever get reassigned. But I'd be happy to use a scope-capturing trick like `partial` if that would set your mind at ease!